zilencer/views.py

```diff
@@ -523,6 +523,8 @@ def truncate_payload(payload: Dict[str, Any]) -> Dict[str, Any]:
         data={
             "total_android_devices": len(android_devices),
             "total_apple_devices": len(apple_devices),
+            "deleted_android_tokens": deleted_android_tokens_list,
+            "deleted_apple_tokens": deleted_apple_tokens_list,
```

Comment:
A seemingly simpler approach could be to scrap this, and instead re-query here for all the registration that the bouncer **does** have and return those. And the self-hosted server deletes any registration that it has that's **not** in this list. But that can race with a new registration by the user probably? So that sounds possibly harder to get correct in the end, and the effect of the race would be pretty annoying :neutral_face:

Reply:
Yeah I think this is the cleaner API.

Reply:
Hmm, actually, thinking about this a bit more: This API does have the problem that if the remote server misses the response where the bouncer deleted a token (say due to network interruption), then the remote server may end up stuck with a stale copy of the token forever.

I suppose an alternative API would be to send the list of pairs of `(app_id, token)` that the remote server expects to be the recipients when sending a push notification, and the bouncer tells it which of those no longer exist (queried after the deletions that happen inline in the request).